src/components/datepicker/localeUtils.ts

```diff
@@ -87,6 +87,15 @@ export const formatDate = (
 export const parseDate = (string: string, format = DateTime.DATE_SHORT as string, locale = defaultLocale) =>
   DateTime.fromFormat(string, format, { locale }).toJSDate();
 
+export const parseMultiFormatsDate = (value: string, formats: string[], locale: string) =>
+  formats.reduce((previous: Date | null, current) => {
```

Comment:
A bit weird to use a reduce here. Why not use a normal `for` loop and do an early return once you have a valid date?